primitives/src/vid/advz.rs

```diff
@@ -388,20 +278,10 @@ where
                 .collect::<Result<_, _>>()
                 .map_err(vid)?,
             all_evals_digest: all_evals_commit.commitment().digest(),
-            elems_len,
-        };
-
-        let commit = {
-            let mut hasher = H::new();
-            for poly_commit in common.poly_commits.iter() {
-                // TODO compiler bug? `as` should not be needed here!
-                (poly_commit as &P::Commitment)
-                    .serialize_uncompressed(&mut hasher)
-                    .map_err(vid)?;
-            }
-            hasher.finalize()
+            bytes_len,
         };
 
+        let commit = Self::poly_commits_hash(common.poly_commits.iter())?;
         let pseudorandom_scalar = Self::pseudorandom_scalar(&common)?;
```

Comment:
Can we derive `pseudorandom_scalar` from `commit` + `all_evals_digest`? There can be redundant hash calls if we derive from `poly_commits`.

Reply:
Yep. Done in commit 8f2240b3f9d01c29f77142c8b2dd8698c0be79fe